src/http/grpc.rs

```diff
@@ -0,0 +1,180 @@
+    metadata::MetadataValue,
+    service::Interceptor,
+    transport::{Channel, Endpoint},
+    Request, Status,
+};
+use tracing::{debug, info, warn};
+
+pub mod tidybee_events {
+    tonic::include_proto!("tidybee_events");
+}
+
+// region: --- Interceptors
+
+pub struct AuthInterceptor {
+    agent_uuid: String,
+}
+
+impl Interceptor for AuthInterceptor {
+    fn call(&mut self, mut request: tonic::Request<()>) -> Result<Request<()>, Status> {
+        debug!(
+            "Adding authorization header to gRPC request: {}",
+            self.agent_uuid
+        );
+
+        match MetadataValue::from_str(format!("Bearer {}", self.agent_uuid).as_str()) {
+            Ok(auth_header) => {
+                request.metadata_mut().insert("authorization", auth_header);
+                Ok(request)
+            }
+            Err(e) => {
+                return Err(Status::internal(format!(
+                    "Failed to create authorization header: {}",
+                    e
+                )));
+            }
+        }
+    }
+}
+
+// endregion: --- Interceptors
+#[allow(private_interfaces)]
+pub struct GrpcClient {
+    pub client: Option<
+        TidyBeeEventsClient<
+            tonic::service::interceptor::InterceptedService<Channel, AuthInterceptor>,
+        >,
+    >,
+    agent_uuid: Option<String>,
+    endpoint: Endpoint,
+}
+
+impl GrpcClient {
+    pub fn new(grpc_server_config: &GrpcServerConfig) -> Self {
+        let endpoint = Channel::from_shared(format!(
+            "{}://{}:{}",
+            grpc_server_config.protocol, grpc_server_config.host, grpc_server_config.port
+        ))
+        .expect("Failed to create endpoint"); // TODO: Handle error
+
+        Self {
+            client: None,
+            agent_uuid: None,
+            endpoint,
+        }
+    }
+
+    #[inline]
+    pub fn set_agent_uuid(&mut self, agent_uuid: &String) {
+        info!("Setting agent uuid for gRPC client: {}", agent_uuid);
+        self.agent_uuid = Some(agent_uuid.clone());
+    }
+
+    // Connect before setting interceptors !
+    pub async fn connect(&mut self) {
+        if self.agent_uuid.is_none() {
+            panic!("Agent UUID is not set for gRPC client");
+            // TODO handle error
+        }
+
+        let channel = match self.endpoint.connect().await {
+            Ok(channel) => channel,
+            Err(e) => {
+                // TODO: Manage connection failure
+                panic!("Failed to connect to gRPC server: {}", e);
+            }
+        };
+        info!("Connected to gRPC server");
+        let interceptor = AuthInterceptor {
+            agent_uuid: self.agent_uuid.clone().unwrap(),
+        };
+        self.client = Some(TidyBeeEventsClient::with_interceptor(channel, interceptor));
+    }
+
+    pub async fn send_create_events_once(&mut self, events: Vec<FileInfo>) {
+        if self.client.is_none() {
+            panic!("gRPC client is not connected");
+            // TODO handle error
```

Comment:
Don't you forget ? Maybe waiting for some code from @majent4, i'll create issue to think about implemting error management here and for each occurences

Reply:
Yes, it was intentional indeed. It's gonna be integrated either in this PR or in the next one.


Thx for raising the issue 🙏